### ISSUE TITLE
Unable to restore custom metric.

### ISSUE THREAD

**Issue by saddamhijazi** [tensorflow/tensorflow#53606]
I have written a code for training a neural network in tensorflow where I added a custom metric just for the sake of printing the current learning rate at each epoch (this is useful for me when I use learning schedules). I trained my model and then I tried to load it again for re-training however I got the following error:

Unable to restore custom metric. Please ensure that the layer implements `get_config` and `from_config` when saving. In addition, please use the `custom_objects` arg when calling `load_model()`.

I tried the proposed solution in many other posts in which one use the custom_objects for indicating the custom metric, however I have not been able to solve this issue, below a simple code to reproduce the issue.

```python

import numpy as np
import tensorflow as tf
from tensorflow import keras


print(tf.__version__)

def get_lr_metric(optimizer):
    def lr(y_true, y_pred):
        return optimizer._decayed_lr(tf.float32) # I use ._decayed_lr method instead of .lr
    return lr


_input = tf.keras.layers.Input(shape=(500), name="fbank") # B*T*F*c
out = tf.keras.layers.Dense(50, activation="tanh")(_input)
probabilities = tf.keras.layers.Dense(2, activation="softmax")(out)
model = tf.keras.Model(inputs=_input, outputs=probabilities)
optimizerAdam = keras.optimizers.Adam(learning_rate=1e-5)
lr_metric = get_lr_metric(optimizerAdam)

model.compile(optimizerAdam, loss=tf.keras.losses.CategoricalCrossentropy(), 
              metrics=['accuracy', lr_metric])

model.summary()

x=np.random.rand(300,500)
y=np.random.rand(300,2)
model.fit(x,y,batch_size=100, epochs=2)

path = 'saved_model/'
model.save(path, save_format='tf')

del model
model = tf.keras.models.load_model('saved_model', compile=False,custom_objects={"lr_metric": lr_metric})
print("Done")


```

**Comment by sushreebarsa** [comment]
@saddamhijazi 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)
Thank you!

**Comment by saddamhijazi** [comment]
Thank you, the issue is posted on the keras project.
